### ISSUE TITLE
Settings controller, CI 3

### ISSUE THREAD

**Issue by mwhitneysdsu** [ci-bonfire/Bonfire#1106]
This is primarily for me to track the problems I'm attempting to fix with this particular part of the site. Currently, I am not aware of any other outstanding issues in CI 3, but any areas which I have not worked on in the last couple of weeks may still be untested.

In CI 3 when attempting to save the settings (from SITE_AREA/settings/settings):
- validation error: The Language field must be an array
- after fixing above, any changes will cause the first attempt to save the page to work, but the page will display an unhelpful error message and all of the settings will be set to their previous values. 
    - Attempting to save the page again will result in the same unless you change the settings to those which were saved the previous time.

In other words, the page saves successfully (once you remove the is_array validation from the language field), but displays an error message and displays the previous (no longer current in the database) settings.

The page will only display a success message if the settings you are attempting to save are the same as the settings in the database.
